packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphThrottleBlock.ts

```diff
@@ -0,0 +1,33 @@
+import type { FlowGraphContext } from "../../../flowGraphContext";
+import type { FlowGraphDataConnection } from "../../../flowGraphDataConnection";
+import { RichTypeNumber } from "../../../flowGraphRichTypes";
+import type { FlowGraphSignalConnection } from "../../../flowGraphSignalConnection";
+import { FlowGraphWithOnDoneExecutionBlock } from "../../../flowGraphWithOnDoneExecutionBlock";
+
+export class FlowGraphThrottleBlock extends FlowGraphWithOnDoneExecutionBlock {
+    public readonly duration: FlowGraphDataConnection<number>;
+    public readonly timeRemaining: FlowGraphDataConnection<number>;

```

Comment:
why is timeRemaining a data connection? to keep everything at the context?

Reply:
Because the specification here: https://docs.google.com/document/d/1MT7gL-IEn_PUw-4XGBazMxsyqsxqgAVGYcNeC4Cj_9Q/edit#heading=h.91x3wearm2v0 requests timeRemaining to be an output flow